addons/web/controllers/main.py

```diff
@@ -1047,8 +1047,12 @@ def content_image(self, xmlid=None, model='ir.attachment', id=None, field='datas
         elif status != 200 and download:
             return request.not_found()
 
-        height = int(height or 0)
-        width = int(width or 0)
+        if headers and 'svg' in dict(headers)['Content-Type']:  # we shan't resize svg images
```

Comment:
I would do: `dict(headers).get('Content-Type', '') == 'image/svg+xml'`

So we match the exact MIME Type, and we are a bit more flexible about what `headers` contains (although it is supposed to contain `headers` following https://github.com/odoo/odoo/blob/11.0/odoo/addons/base/ir/ir_http.py#L346)

Reply:
You could also test that we actually have an image mimetype  (`content_type.startswith('image/')`) to avoid problems with all non-image files, not just neutered SVGs.

An alternative would be to make `tools.image_resize_image()` safer. If `Image.open()` crashes we know it's a file that we can't handle, and we could just return it unmodified.

BTW, the sanitizing question looks irrelevant here, we simply should not try to resize non-supported files, SVG or text.

Reply:
> BTW, the sanitizing question looks irrelevant here, we simply should not try to resize non-supported files, SVG or text.

Indeed, but what we (Tecnativa) want to achieve is to support SVG. Is it a lost battle?

Reply:
Basically true svg support would indeed mean:
- check any image entry point for sanitization in case of svg
- rewrite any resize/crop/etc method to check for svg, in which case no resizing should occur. We might have to manually set the width and height parameter in the svg. There are other things there that might have to be modified accordingly, such as viewBox or enable-background, one would have to check the svg standard to do that properly.
All this is clearly outside of bugfix scope, but it doesn't seem like a very long task. You could probably make a feedback for this feature.

Reply:
@pedrobaeza What do you want to do? Implement sanitizing of SVGs? that seems like a useless risk to take, and a brittle approach. What is the use case?
We could simply extend the current permissions to allow all admins and website editors to upload SVGs without neutering, and that should be enough to cover the important use cases, shouldn't it?

Reply:
My purpose is to be able to handle SVG images everywhere:

- Image widget
- As resources for website

Reply:
@odony should we go ahead then with this?

Reply:
@pedrobaeza Hmm, go ahead with what exactly? What do you want to change concretely, and why do you consider that the current behavior is not sufficient? (sorry if that's obvious to you  ;-), I was under the impression that SVG files were already supported everywhere, provided that the file is uploaded by a trusted user)

Reply:
@odony 
- No, it's not a trusted user but only admin. 
- More importantly it doesn't work in most places because there are things like an automatic conversion to a smaller size at file upload. 

I haven't fixed that because that's way outside of the bug report scope, and then it would require much more testing to be sure that everything is fine everywhere. That sounds more like a task.

So I've updated the commit title to reflect reality: it only works for admin posting an image in the chatter. Anyway see my comment above for what would need to be done.

Use case would be to be able to have svg images for products images for instance. I don't know what fraction of the userbase is really put off by rasterising their vector files though.

Reply:
- *trusted users* was a generic way of saying that we restrict SVG upload, to whom exactly could still be discussed, as long as the principle remains. @pedrobaeza this is the part I insist we should keep in master.
- How many different image processing functions do we have? I count only a half-dozen occurrences of `Image.open()` outside of `tools.image`, and several of them could be delegated to `tools.image`. That does not look like a lot of work. @pedrobaeza for sure you could try to clean this up in master. Supporting SVG (i.e., not crashing) in places where we support other images seems good.
- If further fine-tuning is needed in other places to make some features more SVG friendly, why not, as long as we can keep these changes minimal and clean.

Reply:
May main goal for achieving SVG support is the company logo. IMO only this worths for adding the SVG support. Having it on SVG format, we optimize the size of all PDF documents that contains it, while not sacrificing the quality if very big (obviously, this depends on the logo complexity, but most of the times, it's worth it more than the transparent PNG equivalent).

Another use case is for having SVG images on website. I've just made a test on latest master branch and when inserting an SVG image this is what I get:

![seleccion_007](https://user-images.githubusercontent.com/7165771/40239654-f15189d8-5ab6-11e8-86e7-72017bee16c1.png)

we have a clickable icon for downloading it, but not being displayed.

Reply:
@pedrobaeza so, according to the discussion we had and @odony 's comments, lets start with using `tools.image` everywhere it is missing. Regarding the website, we need to analyse exactly where it is breaking but, no problem to allow it, but one bug at a time.

Reply:
Here you have the needed changes for allowing full SVG support as stated: #26635